docs/ratt-transform/index.md

```diff
@@ -675,84 +675,384 @@ app.use(
+
+RATT is able to access specific elements from such lists based on their *index* or position.  Following the standard practice in Computer Science, RATT refer to the first element in the list as having index 0.  The second element has index 1, etc.
+
+For the above example record, we can assert the first label of the country:
+
+```ts
+app.use(
+  mw.addQuad(
+    mw.toIri('data.countries[0].id', {prefix: prefix.country}),
+    rdfs.label,
+    mw.toLiteral('data.countries[0].name', {language: 'en'})),
+)
+```
+
+This results in the following assertion:
+
+```trig
+country:nl rdfs:label 'The Netherlands'@en.
+```
+
+We can also assert the second label of the same country.  Notice that only the index is different (‘1’ instead of ‘0’):
+
+```ts
+app.use(
+  mw.addQuad(
+    mw.toIri('data.countries[1].id', {prefix: prefix.country}),
+    rdfs.label,
+    mw.toLiteral('data.countries[1].name', {language: 'en'})),
+)
+```
+
+This results in the following assertion:
+
+```trig
+country:de rdfs:label 'Germany'@en.
+```
+
+### Iterating over lists
+
+In the previous section, we saw that we were able to make a first assertion for the first label of the country in the source data.  We were then also able to make a second assertion for the second label of the same country.
+
+But what do we do if there are 100 labels in 100 different languages?  And what do we do if some countries have a label in 2 locales, but other countries have a label in 1 or 3 languaes?  What we need is a simple way to express that we want RATT to make an assertion for each element in the list.
+
+RATT uses the `mw.forEach` function for this purpose.  The following code snippet asserts *each* label for the country in the example data:
+
+```ts
+app.use(
+  mw.forEach('data.countries',
+    mw.addQuad(
+      mw.toIri('id', {prefix: prefix.country}),
+      rdfs.label,
+      mw.toLiteral('name', {language: 'en'}))),
+)
+```
+
+Notice the following details:
+- `mw.forEach` uses the path expression `'data.countries'` to identify the list.
+- Inside the `mw.forEach` function, each element in the list is made available separately.
+- This allows the `'id'` and `'name'` keys to be identified directly.
+
+The above code snippet makes one assertion for every element in the `"countries"` list:
+
+```trig
+country:nl rdfs:label 'The Netherlands'@en-us,
+                      'Nederland'@nl-nl.
+```
+
+The elements that `mw.forEach` iterates over are themselves RATT records.  This implies that all functions that work for full RATT records also work for the RATT records inside `mw.forEach`.
+
+### Special keys when iterating over lists
+
+In the previous section we saw that `mw.forEach` allows us to iterate over the elements in an list.  The elements are themselves made available as RATT records.  The RATT records inside an `mw.forEach `function are smaller.  This allow the regular keys of the iterated-over elements to be accessed directly.
```

Comment:
'over objects in an array' instead of 'elements in a list'.

Reply:
Merged this section with the section called "Iterating over lists of objects".